docs/examples/test_examples.py

```diff
@@ -0,0 +1,268 @@
+""" Tests for the files in nameko/docs/examples
+"""
+import os
+
+import eventlet
+from mock import call, patch
+import pytest
+import requests
+
+from nameko.containers import ServiceContainer
+from nameko.standalone.events import event_dispatcher
+from nameko.standalone.rpc import ServiceRpcProxy
+from nameko.testing.services import entrypoint_waiter
+from nameko.testing.websocket import make_virtual_socket
+
+
+@pytest.yield_fixture
+def container_factory(rabbit_config):
+
+    all_containers = []
+
+    def make_container(service_cls, config, worker_ctx_cls=None):
+        container = ServiceContainer(service_cls, config, worker_ctx_cls)
+        all_containers.append(container)
+        return container
+
+    yield make_container
+
+    for c in all_containers:
+        try:
+            c.stop()
```

Comment:
Yeah, this was lifted from ``conftest.py`` which still uses ``stop`` (which we should also fix). c.f. my note about hoisting the implementations so we can import them here.